src/AWS.Deploy.CLI/ServerMode/Tasks/DeployRecommendationTask.cs

```diff
@@ -30,6 +30,30 @@ public async Task Execute()
             await _orchestrator.DeployRecommendation(_cloudApplication, _selectedRecommendation);
         }
 
+        /// <summary>
+        /// Generates the CloudFormation template that will be used by CDK for the deployment.
+        /// This involves creating a deployment bundle, generating the CDK project and running 'cdk diff' to get the CF template.
+        /// This operation returns the CloudFormation template that is created for this deployment.
+        /// </summary>
+        public async Task<string> GenerateCloudFormationTemplate()
+        {
+            if (_orchestrator._session == null)
+                throw new FailedToCreateCDKProjectException(DeployToolErrorCode.FailedToCreateCDKProject, $"We could not create a CDK deployment project due to a missing dependency '{nameof(_orchestrator._session)}'.");
+            if (_orchestrator._cdkProjectHandler == null)
+                throw new FailedToCreateCDKProjectException(DeployToolErrorCode.FailedToCreateCDKProject, $"We could not create a CDK deployment project due to a missing dependency '{nameof(_orchestrator._cdkProjectHandler)}'.");
+
+            await CreateDeploymentBundle();
+            var cdkProject = await _orchestrator._cdkProjectHandler.ConfigureCdkProject(_orchestrator._session, _cloudApplication, _selectedRecommendation);
+            try
+            {
+                return await _orchestrator._cdkProjectHandler.PerformCdkDiff(cdkProject, _cloudApplication);
```

Comment:
From the CLI project we should not be accessing internal members of the `Orchestrator`. I actually missed that we did in a previous PR made these fields internal. That was probably shouldn't have done.

Reply:
Fixed